src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzer.cs

```diff
@@ -144,7 +144,13 @@ public override void Initialize(AnalysisContext analysisContext)
                 }
 
                 // Compute code metrics.
-                var computeTask = CodeAnalysisMetricData.ComputeAsync(compilationContext.Compilation, compilationContext.CancellationToken);
+                var inheritanceExcludedTypes = compilationContext.Options.GetInheritanceExcludedTypeNamesOption(CA1501Rule, compilationContext.Compilation, compilationContext.CancellationToken);
+
+                var metricsAnalysisContext = new CodeMetricsAnalysisContext(compilationContext.Compilation, compilationContext.CancellationToken)
+                {
+                    IsExcludedFromInheritanceCountFunc = namedType => inheritanceExcludedTypes.Contains(namedType)
+                };
+                var computeTask = CodeAnalysisMetricData.ComputeAsync(metricsAnalysisContext);
```

Comment:
I have decided not to pass the options to the metric classes but instead to use a `Func` on some metrics context class I have created. I did so because I couldn't find any cache for the reading of the options.